src/configuration_server.erl

```diff
@@ -0,0 +1,160 @@
+%%%-------------------------------------------------------------------
+%%% @doc
+%%% This module implements the server, that reads and stores the
+%%% configuration for TEP. It presents an interface for other
+%%% processes to access the configuration variables.
+%%% @end
+%%%-------------------------------------------------------------------
+-module(configuration_server).
+
+-behaviour(gen_server).
+
+-include("configuration_server.hrl").
+
+%%% Export these internal functions if eunit is testing this module
+-ifdef(EUNIT).
+
+-export([default_directories/0, default_filenames/0, choose_config_file/2]).
+
+-endif.
+
+%% API
+-export([start_link/0, get_default_directories/0, get_config_file/0]).
+
+%% gen_server callbacks
+-export([init/1, handle_call/3, handle_cast/2, handle_info/2, terminate/2, code_change/3]).
+
+-define(SERVER, ?MODULE).
+
+-record(configuration, {config_file_path :: file:filename_all()}).
+
+%%%===================================================================
+%%% API
+%%%===================================================================
+%% @doc Spawns the server and registers the local name (unique)
+-spec start_link() ->
+    {ok, Pid :: pid()} | ignore | {error, Reason :: term()}.
+start_link() ->
+    gen_server:start_link({local, ?SERVER}, ?MODULE, [], []).
+
+%% @doc Returns a list with the default configuration file locations
+-spec get_default_directories() ->
+    {ok, [file:filename_all()]}.
+get_default_directories() ->
+    gen_server:call(configuration_server, default_directories).
+
+%% @doc Returns the used configuration file
+-spec get_config_file() ->
+    {ok, file:filename_all()} | {error, Reason :: term()}.
+get_config_file() ->
+    gen_server:call(configuration_server, config_file).
+
+%%%===================================================================
+%%% gen_server callbacks
+%%%===================================================================
+%% @private
+%% @doc Initializes the server
```

Comment:
```suggestion
%%%===================================================================
%%% gen_server callbacks
%%%===================================================================

%% @private
%% @doc Initializes the server
```

Reply:
This will be automatically removed by the formatter `rebar3 fmt`.